Feature 550 v3.2.0

bikegeek opened the pull request:
## Pull Request Testing ##

- [ ] Describe testing already performed for these changes:</br>

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes or No]**

- [ ] Do these changes include sufficient testing updates? **[Yes or No]**

- [ ] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Do these changes introduce new SonarQube findings? **[Yes or No]**</br>
If **yes**, please describe:

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **METplotpy-X.Y.Z Development** project for official releases
- [ ] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

bikegeek committed:
Updated the release_date

bikegeek committed:
Removed the -rc1 suffix for coordinated release

bikegeek committed:
Updated for coordinated release

bikegeek committed:
Rotate authorship for coordinated release

bikegeek committed:
Update year of copyright date

jprestop committed:
Per #550, modify formatting of release notes header

jprestop committed:
Per #550, attempting to resolve SonarQube error with regard to coverage

Add source and omit settings for coverage tool

jprestop committed:
Per #550, correcting syntax

Updated source paths for coverage tool and adjusted omit settings.

jprestop committed:
Per #550, update coverage omit pattern for _netCDF4.pyx

jprestop committed:
Per #550, another attempt to resolve the SonarQube error

jprestop committed:
Per #550, another attempt to resolve the SonarQube error

Updated coverage configuration to omit specific files.

jprestop commented:
@bikegeek The SonarQube scans are failing, seemingly due to:

> No source for code: 'test/histogram_2d/src/netCDF4/_netCDF4.pyx'; see https://coverage.readthedocs.io/en/7.11.3/messages.html#error-no-source
Error: Process completed with exit code 1.

Based on the information in the link, I've tried to modify the pyproject.toml file to get rid of the error.  Take a look at my modifications when you get a chance. I still have not found a solution. Are you familiar with the pyproject.toml file? If so, any ideas on how to resolve the error?  I'm going to move on to the other releases with the hope that you'll know better how to resolve this problem.

bikegeek committed:
Update pyproject.toml

move the omit from the tool.coverage.report to the tool.coverage.run to see if the histogram_2d/src/netCDF4 can be omitted from the trace

bikegeek committed:
Update pyproject.toml

try [run] source=. to prevent measuring code outside of project

bikegeek committed:
remove the [run] block, this causes issues with another GHA

bikegeek committed:
add source=. to only check the METplotpy source code and not third party libraries

bikegeek committed:
Use different syntax to omit the histogram_2d/src from report

bikegeek committed:
Return to original state, moved omit to the .coveragec config

bikegeek committed:
Update pyproject.toml

remove extraneous line

bikegeek committed:
Update release-notes.rst

bikegeek committed:
Update conf.py

jprestop reviewed (approved):
@bikegeek Thank you so much for figuring out the problem! I approve this request.
